hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,93 @@
+    
+## Description
+
+Though the recommendations in AIS31 are based around broad functional requirements, the recommendations in SP 800-90 are very prescriptive in nature, outlining the exact constructs needed for approval.
+Thus the interface and implementation are largely driven by these explicit constructs, particularly the CTR_DBRG construct.
+
+
+The CSRNG IP consists of four main components:
+1. An AES primitive
+2. The CTR_DRBG state-machine (CTR_DRBG_FSM) which drives the AES primitive, performing the various encryption sequences prescribed for approved DRBGs in SP 800-90A.
+These include:
+
+    1. **The Derivation Function:**  
+       Part of the instantiation and reseed routines, this routine assembles the previous seed material (on reseed only), application inputs, and entropy.
+    2. **The Instantiation Routine:**
+       Combines application inputs, external entropy and nonce (more entropy) via the derivation function.
+    3. **The Reseed Routine:**
+       Combines the previous seed material with external entropy to generate a new seed.
+    4. **The Generate Routine:**
+       Generates up to CSRNG_MAX_GENERATE random bits.
+       If called with prediction_resistance_flag, forces a reseed.
+     5. **The Update Routine:**
+        Updates the internal state of the DRNG instance after each generate call.
+3. State vectors for each DRNG instance.
+4. Interface logic and access control for each instance.
+
+## Note on the term "Entropy"
+
+Every DRNG requires some initial seed material, and the requirements for the generation of that seed material varies greatly between standards, and potentially between CC security targets.
+In all standards considered, DRNG's require some "entropy" from an external source to create the initial seed.
+However, the rules for obtaining said entropy differ.
+Furthermore the required delivery mechanisms differ.
+For this reason we must make a clear distinction between "Physical" (or "Live" or "True") entropy and "Factory Entropy".
+This distinction is most important when considering the creation of IP which is both compatible with the both the relatively new SP 800-90 recommendations, as well as the well-established FIPS 140-2 guidelines.
+
+- Physical entropy is the only type of "entropy" described in SP 800-90.
+The means of generation is described in [SP 800-90B](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90B.pdf).
+One statistical test requirement is that physical entropy must be unique between reboot cycles, ruling out sources such as one-time programmable (OTP) memories.
+In SP 800-90A, the delivery mechanism must come through a dedicated interface and "not be provided by the consuming application".
+
+- Factory entropy is a type of entropy described in the [FIPS 140-2 implementation guidance (IG)](https://csrc.nist.gov/csrc/media/projects/cryptographic-module-validation-program/documents/fips140-2/fips1402ig.pdf) section 7.14, resolution point 2(a).
```

Comment:
Just to confirm, will software have access to factory_entropy mode in instance 0, Or does this require a second instance?

Reply:
I may have to defer this to @mwbranstad, but my understanding is that everything in instance 0 is accessible from TL-UL, so you could instantiate that instance in this mode, use it, and then scrap it, and start another instance (in the same instance 0 slot).  However this would be effectively using two instances.  @mwbranstad would it be possible to make multiple instances SW accessible, and then scrap selected ones using OTP?

Reply:
@martin-lueker it turns out that sw interface can start as many instances as are available in the hw. all modes are available to all instances. BTW, due to an interconnect limitation, the sw instance is not 0 now, but the highest number of available APP interfaces. If there are combinations of more than one sw app interface, we would need to discuss this more.